src/script.c

```diff
@@ -255,7 +255,7 @@ u8 *MapHeaderGetScriptTable(u8 tag)
         if (*mapScripts == tag)
         {
             mapScripts++;
-            return (u8 *)(mapScripts[0] + (mapScripts[1] << 8) + (mapScripts[2] << 16) + (mapScripts[3] << 24));
+            return (u8 *)(*mapScripts + (*(mapScripts + 1) << 8) + (*(mapScripts + 2) << 16) + (*(mapScripts + 3) << 24));
```

Comment:
Why did you do this?

Reply:
Yes, please explain the reasoning behind this commit.  It is wholly unnecessary and makes the code more difficult to read..

Reply:
Because it is accessing a memory location, not an array, which the original code was misleading about.

Reply:
And you are already checking the value of the map scripts pointer already, so consistency too